coap/adapter.go

```diff
@@ -1,61 +1,257 @@
-package adapter
+package coap
 
 import (
 	"encoding/json"
-	"log"
+	"errors"
+	"fmt"
 	"net"
+	"strings"
 
-	"github.com/dustin/go-coap"
+	"encoding/base64"
+
+	mux "github.com/dereulenspiegel/coap-mux"
+	coap "github.com/dustin/go-coap"
+	"github.com/go-kit/kit/log"
 	"github.com/mainflux/mainflux"
+	manager "github.com/mainflux/mainflux/manager/client"
 	broker "github.com/nats-io/go-nats"
-	"go.uber.org/zap"
 )
 
-const protocol string = "coap"
+const (
+	token    string = "token"
+	prefix   string = "src.coap."
+	channel  string = "channel_id"
+	protocol string = "coap"
+)
+
+var (
+	errUnauthorizedAccess error = errors.New("missing or invalid credentials provided")
+)
 
+// Observer struct provides support to observe message types.
 type Observer struct {
 	conn    *net.UDPConn
 	addr    *net.UDPAddr
 	message *coap.Message
+	sub     *broker.Subscription
 }
 
-type CoAPAdapter struct {
+// Subscriber represents token-NATS subscription map. Token is represented as string in
+// order to simplify access to subscription as well as to simplify unsubscribe feature.
+type Subscriber map[string]*Observer
+
+// AdapterService struct represents CoAP adapter service.
+type AdapterService struct {
 	obsMap map[string][]Observer
```

Comment:
Rename to `observers`. The map suffix is unnecessary since it is clear from the type that it is a map.